frontend/src/tests/FetchDeveloper.test.tsx

```diff
@@ -0,0 +1,40 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { vi, describe, it, expect } from 'vitest';
+import axios from 'axios';
+import FetchDevelopers from '../components/FetchDevelopers';
+
+// Mock the Axios module
+vi.mock('axios');
+
+describe('FetchDevelopers component', () => {
+  it('fetches and displays developers data', async () => {
+    // Mocked data returned by the Axios GET request
+    const developers = [
+      { id: 1, first_name: 'G', last_name: 'Damaso' },
+      { id: 2, first_name: 'K', last_name: 'Caparas' },
+      { id: 3, first_name: 'S', last_name: 'Spy' },
+    ];
+
+    // Setup the mock to return the mocked data
+    vi.mocked(axios.get).mockResolvedValueOnce({ data: developers });
+
+    // Render the component
+    render(<FetchDevelopers />);
+
+    // Wait for the developers data to be fetched and displayed
+    await waitFor(() => {
+      // Check if the correct number of developer rows are displayed
+      const rows = screen.getAllByRole('row');
+      // Check for developers.length
+      expect(rows).toHaveLength(developers.length + 1);
+      // Check if specific developer data is displayed
+      expect(screen.getByText('K')).toBeInTheDocument();
+      expect(screen.getByText('Caparas')).toBeInTheDocument();
+      expect(screen.getByText('G')).toBeInTheDocument();
+      expect(screen.getByText('Damaso')).toBeInTheDocument();
+      expect(screen.getByText('S')).toBeInTheDocument();
+      expect(screen.getByText('Spy')).toBeInTheDocument();
```

Comment:
There's a good opportunity to condense this section and by much more DRY :) 
Here's an example of things you can do to

```js
      ['K', 'Caparas', 'G', 'Damaso', 'S', 'Spy'].forEach((item: string) => {
        expect(screen.getByText(item)).toBeInTheDocument();
      });
```
This performs the  exact same function, but without the repetition! Though if you incorporate this, using a named variable for the array instead of how I just implemented it directly would be a much better practice. 


If you use the second function argument `index` you could probably check that the information is displayed on the correct rows too :)

Reply:
this is what I'm actually looking for. Testing tends to be really repetitive, so I was looking for other ways to make it shorter, and you gave me an idea for a solution. Thanks matthew!

Reply:
I've used this named array that was already declared on my test.

![image](https://github.com/bcgov/nr-sustainment-capstone-2024/assets/144957476/0513c375-1745-493a-8224-da3707eea33e)
![image](https://github.com/bcgov/nr-sustainment-capstone-2024/assets/144957476/189b44f6-2491-482d-80c6-fe13902e5ebc)

Reply:
Now that's DRY :)

Reply:
it works 😄